src/Monitor/Monitor/ScheduledQueryRules/NewScheduledQueryRuleLogMetricTriggerCommand.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Insights.OutputClasses;
+using Microsoft.Azure.Management.Monitor.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Insights.ScheduledQueryRules
+{
+    /// <summary>
+    /// Create a ScheduledQueryRule Metric Trigger object
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRuleLogMetricTrigger"), OutputType(typeof(PSScheduledQueryRuleLogMetricTrigger))]
+    public class NewScheduledQueryRuleLogMetricTriggerCommand : MonitorCmdletBase
+    {
+
+        #region Cmdlet parameters
+
+        [Parameter(Mandatory = true, HelpMessage = "The metric threshold operator : GreaterThan, LessThan, Equal")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("GreaterThan", "LessThan", "Equal")]
```

Comment:
@shpimpal this should be `PSArgumentCompleter` to allow for the same tab functionality that is allowed by `ValidateSet`, but also allows users to provide values outside this set (to immediately support new values as they're unveiled by the API)

This comment can be applied to all instances of `ValidateSet` being used

Reply:
will add `PSArgumentCompleter`. But we do not support values other than the ones in the set.